cmd/kube-apiserver/app/options/validation.go

```diff
@@ -142,11 +141,5 @@ func (s CompletedOptions) Validate() []error {
 		errs = append(errs, fmt.Errorf("--apiserver-count should be a positive number, but value '%d' provided", s.MasterCount))
 	}
 
-	// TODO(#125980): remove in 1.32
-	effectiveVersion := s.GenericServerRunOptions.ComponentGlobalsRegistry.EffectiveVersionFor(s.GenericServerRunOptions.ComponentName)
-	if err := utilversion.ValidateKubeEffectiveVersion(effectiveVersion); err != nil {
```

Comment:
it still seems like we need the validation here, to make sure we're in an acceptable range, even if the range widens to allow more versions

Reply:
I've added back all validation logic, removed the associated TODOs to remove them in 1.32 and updated https://github.com/kubernetes/kubernetes/issues/125980 stating such